From 6f7636cfc6dffb387afe21f4f3bff119f8d8e033 Mon Sep 17 00:00:00 2001
From: Eric Leblond <eric@regit.org>
Date: Thu, 31 Oct 2019 13:29:56 +0100
Subject: [PATCH] ebpf: avoid to include if_tunnel.h

This is causing a dependency issue as file from another architecture
have to be installed.
---
 ebpf/xdp_lb.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/ebpf/xdp_lb.c
+++ b/ebpf/xdp_lb.c
@@ -26,7 +26,6 @@
 /* Workaround to avoid the need of 32bit headers */
 #define _LINUX_IF_H
 #define IFNAMSIZ 16
-#include <linux/if_tunnel.h>
 #include <linux/ip.h>
 #include <linux/ipv6.h>
 #include <linux/tcp.h>
@@ -36,6 +35,12 @@
 
 #include "hash_func01.h"
 
+#define GRE_CSUM        __cpu_to_be16(0x8000)
+#define GRE_ROUTING     __cpu_to_be16(0x4000)
+#define GRE_KEY         __cpu_to_be16(0x2000)
+#define GRE_SEQ         __cpu_to_be16(0x1000)
+#define GRE_VERSION     __cpu_to_be16(0x0007)
+
 #define LINUX_VERSION_CODE 263682
 
 /* Hashing initval */
